src/test/java/uk/gov/pay/directdebit/mandate/resources/MandateResourceIT.java

```diff
@@ -87,70 +93,56 @@ public void tearDown() {
+                .post("/v1/api/accounts/{accountId}/mandates".replace("{accountId}", accountExternalId))
                 .then()
-                .statusCode(Response.Status.CREATED.getStatusCode())
-                .body(JSON_MANDATE_ID_KEY, is(notNullValue()))
-                .body("return_url", is(returnUrl))
-                .body("created_date", is(notNullValue()))
-                .contentType(JSON);
-        MandateExternalId externalMandateId = MandateExternalId.valueOf(response.extract().path(JSON_MANDATE_ID_KEY).toString());
-
-        String documentLocation = expectedMandateLocationFor(accountExternalId, externalMandateId);
-        String token = testContext.getDatabaseTestHelper().getTokenByMandateExternalId(externalMandateId).get("secure_redirect_token").toString();
-
-        String hrefNextUrl = "http://Frontend/secure/" + token;
-        String hrefNextUrlPost = "http://Frontend/secure";
-
-        response
-                .body("links", hasSize(3))
-                .body("links", containsLink("self", "GET", documentLocation))
-                .body("links", containsLink("next_url", "GET", hrefNextUrl))
-                .body("links", containsLink("next_url_post", "POST", hrefNextUrlPost, "application/x-www-form-urlencoded", new HashMap<String, Object>() {{
-                    put("chargeTokenId", token);
-                }}));
-
-        Map<String, Object> createdMandate = testContext.getDatabaseTestHelper().getMandateByExternalId(externalMandateId);
-
-        assertThat(createdMandate.get("external_id"), is(notNullValue()));
-        assertThat(createdMandate.get("return_url"), is(returnUrl));
-        assertThat(createdMandate.get("gateway_account_id"), is(gatewayAccountFixture.getId()));
+                .statusCode(HttpStatus.SC_UNPROCESSABLE_ENTITY)
+                .body("message", contains(expectedErrorMessage))
+                .body("error_identifier", is("GENERIC"));
     }
 
     @Test
-    public void shouldCreateAMandate_withAllFields() throws Exception {
+    @Parameters({"null", "raindrops on roses and whiskers on kittens"})
+    public void shouldCreateAMandateSuccessfully(@Nullable String description) throws Exception {
         String accountExternalId = gatewayAccountFixture.getExternalId();
         String returnUrl = "http://example.com/success-page/";
-        String postBody = new ObjectMapper().writeValueAsString(ImmutableMap.builder()
+        
+        ImmutableMap.Builder createMandateBuilder = ImmutableMap.builder()
                 .put("return_url", returnUrl)
-                .put("service_reference", "test-service-reference")
-                .build());
-
-        String requestPath = "/v1/api/accounts/{accountId}/mandates"
-                .replace("{accountId}", accountExternalId);
+                .put("service_reference", "test-service-reference");
+        
+        if (description != null) createMandateBuilder.put("description", description);
```

Comment:
Its enshrined in our formatting standards that `{}` should always be used around `if` statements.
https://gds-way.cloudapps.digital/manuals/programming-languages/java.html#code-formatting